src/core/pubkey.zig

```diff
@@ -22,8 +22,8 @@ pub const Pubkey = extern struct {
         return .{ .data = bytes[0..size].* };
     }
 
-    pub fn fromPublicKey(public_key: *const std.crypto.sign.Ed25519.PublicKey) Self {
-        return fromBytes(&public_key.bytes) catch unreachable;
+    pub fn fromPublicKey(public_key: *const std.crypto.sign.Ed25519.PublicKey) !Self {
+        return Self.fromBytes(&public_key.bytes);
```

Comment:
I agree, I also think in this context specifically it is safe to 'trust' that the bytes in the Ed25519.PublicKey are valid and handle the error if it ever becomes a real issue.  Given that in release safe unreachable does not produce a particular informative stack trace, perhaps we should use @panic with an informative message to prevent painful debugging down the line if some code which is assumed to be unreachable is encountered?

Reply:
My reasoning is identical to @dnut here. If something isn't truly unreachable then `unreachable` is incorrect.

Reply:
This error is unreachable. But we don't need to use an error here, this code should be:
```zig
    pub fn fromPublicKey() Self {
        return .{ .data = public_key.bytes };
    }
```

Reply:
I think I missread which `fromBytes` function we were calling, thought it was the pubkey's fromBytes which has the `NonCanonical` error.
I can change this back!

Reply:
Changed back here: 617162def8992b3de2087745289baf7e15f8cdfa